cmake/generic.cmake

```diff
@@ -331,3 +331,14 @@ function(go_test TARGET_NAME)
   add_custom_target(${TARGET_NAME} ALL DEPENDS ${TARGET_NAME}_timestamp ${go_test_DEPS})
   add_test(${TARGET_NAME} ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME})
 endfunction(go_test)
+
+function(proto_library TARGET_NAME)
+  set(oneValueArgs "")
+  set(multiValueArgs SRCS)
+  cmake_parse_arguments(proto_library "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+  set(proto_srcs)
+  set(proto_hdrs)
+  protobuf_generate_cpp(proto_srcs proto_hdrs ${proto_library_SRCS})
+  include_directories(${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
I have tested this PR, and it works fine to build the `example_proto` through a simple command:
```
proto_library(example_proto SRCS example.proto)
```
No undefined problem and no dependency problem! Great work!
Just remind here that what kind of including path we want. If we want to use `#include "paddle/framework/example.pb.h"`, we may need to change `CMAKE_CURRENT_BINARY_DIR` to `CMAKE_BINARY_DIR`.